packages/components/src/drop-zone/provider.js

```diff
@@ -212,7 +209,7 @@ class DropZoneProvider extends Component {
 		const { position, hoveredDropZone } = this.state;
 		const dragEventType = getDragEventType( event );
 		const dropZone = this.dropZones[ hoveredDropZone ];
-		const isValidDropzone = !! dropZone && this.container.contains( event.target );
+		const isValidDropzone = !! dropZone && this.container.current.contains( event.target );
```

Comment:
If this is the only instance of the DOM reference, could we just have `onDrop` be bound as an event handler to the wrapping `div`? I guess the main difference is the lack of `stopPropagation` / `preventDefault` on all other drops on the page, which I could see as equally "correct" and "annoying" (in cases of mis-drops causing navigation).

Reply:
Also, with use of Slot/Fill, using DOM `Element#contains` isn't guaranteed to be accurate anyways.

Reply:
If this provider is at the top-level, is there really any point to checking this at all?

Assuming we're tracking hover for the relevant `dropZone`, that seems to be an extra (better?) level of certainty to where the drop is taking place.

Reply:
Yes, there is, this fixes a bug where drag and dropping to the media library creates the image twice. (It's also caught by the dropzones in the editor even if the the dragged item is outside the editor)

Reply:
Do you recall the issue / pull request for it? This sounds a bit fishy...

Makes me wonder if there's possibly something going on with React DevTools double-rendering.

Reply:
So the original PR fixing this issue is this one https://github.com/WordPress/gutenberg/pull/5432

but I recall it being refactored to use the container technique in another PR. Looking

Reply:
Looks like it was introduced here https://github.com/WordPress/gutenberg/pull/4115 but I don't recall the exact reasons :( Will investigate more.

Reply:
So yes, in my testing, it is still necessary to avoid the double upload issue.